### ISSUE TITLE
Freezes with WPA2/3-Enterprise

### ISSUE THREAD

**Issue by Xernium** [opa334/Dopamine#86]
# Important: Read the updates below

## System information:
System version: 15.0.2
Model: iPad Pro (12.9-inch) (5th generation) (M1, 8gb RAM)
Previously jailbroken with Fugu 15 Max: No
Previously jailbroken with Xina: Yes, but fully uninstalled long ago
Notes: Tested each step with smart keyboard & apple pencil attached and detached.

## Description of issue:
The device freezes and userspace restart takes very long. The device shows WIFI as off, but it is not off. Changing any settings is not possible because the user interface freezes for minutes at a time.
Input changing any wireless settings is ignored entirely. All other input takes seconds to minutes to register.

## How to reproduce:
1. Jailbreak with Dopamine
2. Jailbreak succeeds and re-enables wifi
3. Userspace reboot takes thrice as long as a reboot of the device

After userspace reboot the device hangs for seconds to minutes at a time between each input. Apps can be opened but take ages, any changes to wireless settings are ignored. Wifi shows as off but is on. Wifi is also not connecting. Apps in the background crash continuously (like Termius) and uninstalling them does not fix the issue. Disconnecting the iPad accessories does also not fix it.
I have once managed to open the settings app before the device locked up completely

![0C58AC9F-C94B-4EF1-8290-4BF86D2A33FE](https://user-images.githubusercontent.com/10787350/235871980-155efe24-1bf3-426b-9dac-1359d9e03645.png)
After trying for a while the device crashes

## Investigation:
I‘ve attempted to re-disable Wifi before the iPad has a chance to userspace reboot post-exploit.
If successful, the userspace reboot takes only seconds. The device is perfectly usable after (no freezes), including wireless settings. However, the iPad refuses to join any Wifi except my iPhone hotspot, the hotspot works perfectly. 
I should note that the wifi networks in question are WPA2/3-Enterprise. It produces the following error messages as shown:

![659A6227-0E6B-437B-BB60-1FF1B54365E6](https://user-images.githubusercontent.com/10787350/235872154-c68ee9aa-762a-4d52-b687-2480e852d298.png)
![841E8644-A992-43ED-A39C-9F02F0DEC562](https://user-images.githubusercontent.com/10787350/235872178-3d5cdc7f-dafd-4d65-91b6-a1d01c0af859.png)

These Wifi-Networks work perfectly fine while unjailbroken.

I can attach crash logs as necessary; Many services generated them due to watchdog timeouts, so I will only attach the panic-full
[panic-full-2023-05-03-103533.000.txt](https://github.com/opa334/Dopamine/files/11380982/panic-full-2023-05-03-103533.000.txt)

**Comment by Xernium** [comment]
# Update
This issue with freezing seems to be exclusive to WPA2/3-Enterprise and does not occur with any non-enterprise (RADIUS or other) encrypted network. For these the wifi-bug seems to still be present in some capacity

**Comment by opa334** [comment]
This is not the wifi bug, the wifi bug would immediately panic. I don't know what it is though... Might be some process crash looping

**Comment by Xernium** [comment]
Any other crash logs I can provide for you? 
The wifi in question I‘ve had the issue on uses WPA3-Enterprise (transition mode) with RADIUS authentication

**Comment by MarkPLacer** [comment]
Hi, I’m getting the same issue with my school wifi, it’s eduroam also. I first thought it was their issue but I guess not

**Comment by mreser99** [comment]
This exactly happened to me too.

**Comment by opa334** [comment]
Okay so for debugging, can you test Fugu15 Max Beta 8 and check if it happens there too? I will prepare some debug builds of Dopamine in the mean time to help me nail down this issue.

**Comment by opa334** [comment]
Also that panic is unrelated, don't see how this could randomly happen, must have been exploit failure and you sent the wrong one. Regarding the watchdog thing, shouldn't that panic also? Please send me any service crashes you can find.

**Comment by MarkPLacer** [comment]
> Okay so for debugging, can you test Fugu15 Max Beta 8 and check if it happens there too? I will prepare some debug builds of Dopamine in the mean time to help me nail down this issue.

I'm convinced that it doesn't occur on b8. It's true that I had holidays last week, but I believe I was using b8 prior to that, and the school's euduroam wifi was functional.

**Comment by opa334** [comment]
> > Okay so for debugging, can you test Fugu15 Max Beta 8 and check if it happens there too? I will prepare some debug builds of Dopamine in the mean time to help me nail down this issue.
> 
> I'm convinced that it doesn't occur on b8. It's true that I had holidays last week, but I believe I was using b8 prior to that, and the school's euduroam wifi was functional.

That's really weird though, there haven't been that many changes compared to beta 8

**Comment by mreser99** [comment]
I had no problem with enterprise wifi at work with Beta 8, was working fine.

**Comment by opa334** [comment]
Sorry the previous builds where broken, reuploaded them, please test again now

**Comment by salvatore8686** [comment]
Qui nessun problema WPA2

**Comment by opa334** [comment]
> I was also unable to connect to `eduroam` on iPhone 13, 15.1.1, never used Xina or Fugu builds before Dopamine release. NO-FORKFIX solved the issue.

crap, can other people confirm this too?

**Comment by PixelMelt** [comment]
> > I was also unable to connect to `eduroam` on iPhone 13, 15.1.1, never used Xina or Fugu builds before Dopamine release. NO-FORKFIX solved the issue.
> 
> crap, can other people confirm this too?

NO-FORKFIX Also fixed this issue for me.

**Comment by opa334** [comment]
Okay I got some crash logs that confirm this. In some recent commit I made it so that if anything about forkfix fails, it crashes both itself and the child, this seems to cause the issue. I don't know why it fails however.

**Comment by opa334** [comment]
Anyone here that can test builds who is able to access syslog? (Via console.app or idevicesyslog)

**Comment by jcrm1** [comment]
I have idevicesyslog- do you want logs from the normal build, NO-FORKFIX, both, or something else?

**Comment by opa334** [comment]
Nevermind no need for idevicesyslog.

https://www.mediafire.com/file/e26pkkrv6f7d5ns/Dopamine.FORKFIX-LOG.tipa/file

Jailbreak with this build once and trigger the issue, then post `/var/mobile/forkfix.log` here.

**Comment by opa334** [comment]
Okay again please using this build:
https://www.mediafire.com/file/5s3pr6c0vpz7yjm/Dopamine.FORKFIX-LOG-2.tipa/file

**Comment by jcrm1** [comment]
```
forkfix /usr/libexec/configd - returning 1
forkfix /System/Library/PrivateFrameworks/MobileSoftwareUpdate.framework/XPCServices/com.apple.MobileSoftwareUpdate.CleanupPreparePathService.xpc/com.apple.MobileSoftwareUpdate.CleanupPreparePathService - returning 1
forkfix /private/preboot/D38885A738481C990C5F299DCDAD254F8A9DD60B2E11E91E746E5BF53A7EB45A30F048EFC8630503987A0D2F9742BBC8/jb-0u3mLj/procursus/usr/bin/dpkg - returning 0
pid_resume returned -1, errno: 1, description: Operation not permitted
forkfix /usr/libexec/configd - returning 1
pid_resume returned -1, errno: 1, description: Operation not permitted
forkfix /System/Library/PrivateFrameworks/MobileSoftwareUpdate.framework/XPCServices/com.apple.MobileSoftwareUpdate.CleanupPreparePathService.xpc/com.apple.MobileSoftwareUpdate.CleanupPreparePathService - returning 1
```

**Comment by opa334** [comment]
Yeah ok this really sucks, seems like I have to rework forkfix for the forth time.....

**Comment by opa334** [comment]
Okay everyone please test this: https://www.mediafire.com/file/k9m3a0ef5yjggvu/Dopamine.tipa/file

Should properly fix the issue

**Comment by opa334** [comment]
Should be fixed in 1.0.2.
